hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/HAUtils.java

```diff
@@ -387,7 +387,16 @@ public static File getMetaDir(DBDefinition definition,
               waitForCACerts(certClient::updateCAList, expectedCount),
               waitDuration);
         } else {
-          caCertPemList = certClient.listCA();
+          List<String> x509Certificates = new ArrayList<>();
+          if (certClient.getRootCACertificate() != null) {
+            x509Certificates.add(CertificateCodec.getPEMEncodedString(
+                certClient.getRootCACertificate()));
+          }
+          if (certClient.getCACertificate() != null) {
+            x509Certificates.add(CertificateCodec.getPEMEncodedString(
+                certClient.getCACertificate()));
+          }
```

Comment:
Can we extract a method and reuse it here in and `!SCMHAUtils.isSCMHAEnabled` case?

Reply:
Ya just found out, and did that